test/core/util/port_isolated_runtime_environment.cc

```diff
@@ -43,14 +43,26 @@ static int get_random_port_offset() {
 static int s_initial_offset = get_random_port_offset();
 static gpr_atm s_pick_counter = 0;
 
-int grpc_pick_unused_port_or_die(void) {
+static int pick_unused_port_or_die(void) {
   int orig_counter_val =
       static_cast<int>(gpr_atm_full_fetch_add(&s_pick_counter, 1));
   GPR_ASSERT(orig_counter_val < (MAX_PORT - MIN_PORT + 1));
   return MIN_PORT +
          (s_initial_offset + orig_counter_val) % (MAX_PORT - MIN_PORT + 1);
 }
 
+int grpc_pick_unused_port_or_die(void) {
+retry:
```

Comment:
Other than `goto`, how about using while (true) { .. return port; } ?

Reply:
you're right, much better. I wanted to look fancy ;-)